aws/src/main/java/org/apache/iceberg/aws/glue/LockManager.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+interface LockManager {
+
+  /**
+   * Try to lock a table once
+   * @param database Glue database name
+   * @param table Glue table name
+   * @param expireMillis max duration to hold the lock in milliseconds
+   * @return if lock acquisition succeeded or not
+   */
+  boolean tryLock(String database, String table, long expireMillis);
```

Comment:
Can we extend this interface a little bit? Again to help us w/ Nessie. Being able to pass at the very least the new metadata location would broaden the types of actions the implementer of this class can do.

Similar: Could we call this `ConcurrencyManager` or something similar? To hihglight that this is helping negotiate commit-ability rather than a pure lock mechanism?

Reply:
Could you provide more details for what is needed in Nessie? I have not read too much into the Nessie package yet (will do during thanksgiving), but one thing I think make more sense for both use cases is to expose the interface like 

```
boolean tryLock(String lockId, long expireMillis);
````

And then formulate the lock id based on different needs, does that work in your case?

Reply:
Sorry for the delay @jackye1995 . I've been thinking a lot about this and its not straigthtforward to do consistently. If nessie acts as the lock then acquiring the lock is effectively commiting the update to Nessie. A failure in the Glue layer would then require a rollback of nessie or the table is in an inconsistent state across Nessie/Glue. Obviously thats not acceptable, but possible. Let me think more and I can update you on Monday.

I like your suggestion re `lockId`. Doing a String may not be quite right as it might need excessive parsing by the impl. Nessie specifically needs at minimum `TableIdentifier` (or some derivation of it) and the current metadata location. I guess dynamo needs simply the `TableIdentifier`. So we could add the metadata location to the interface or we can collapse into a lock id object of some sort? The `unlock` could then take that object?

Reply:
For now I have updated the interface to take a Sting `lockId`. I am curious about how exactly are you thinking about integrating Nessie with Glue, so I am just leaving the interface to be as open as possible for easier integration in the future. The use of an object `LockId` seems a bit convoluted, since the interface was intended to be used as an internal lock mechanism for Glue catalog, but based on your description it's like Glue and Nessie doing duplicated work. Let's discuss this in slack and have another PR for the actual work if necessary.